trieste/utils/mo_objectives.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains toy multi-objective functions, useful for experimentation.
+"""
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from functools import partial
+
+import tensorflow as tf
+
+from ..type import TensorType
+
+
+class MultiObjectiveTestProblem(ABC):
+    """base class for synthetic multi-objective test functions"""
+
+    @abstractmethod
+    def prepare_benchmark(self):
+        """
+        Evaluate original problem.
+        """
+
+    @abstractmethod
+    def gen_pareto_optimal_points(self, n: int):
```

Comment:
This also needs a return type, and ideally a :param parameter description in the docstring.

And again, is "gen" necessary?

Reply:
Type and doc added. 
For the "gen", if speaking from the sampling of distribution perspective maybe it is good to say so?  (Actually this name is borrowed form [here](https://github.com/pytorch/botorch/blob/5a5173b6ad89fd52aa66e5ed942208b25ea4b03b/botorch/test_functions/multi_objective.py#L188)).  
While indeed not all the Pareto optimal points are "generated" (as for instance the VLMOP2's are deterministically chosen), hope at least for now it makes some sense, please point out if it is better to remove still.

Reply:
Happy to leave it in.